File: src/cs/AddIns/AddInDemo/DraftRibbon.cs
using SolidEdgeCommunity.AddIn;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace AddInDemo
{
    class DraftRibbon : SolidEdgeCommunity.AddIn.Ribbon
    {
        const string _embeddedResourceName = "AddInDemo.DraftRibbon.xml";

        public DraftRibbon()
        {
            // Get a reference to the current assembly. This is where the ribbon XML is embedded.
            var assembly = System.Reflection.Assembly.GetExecutingAssembly();

            // In this example, XML file must have a build action of "Embedded Resource".
            this.LoadXml(assembly, _embeddedResourceName);
        }

        public override void OnControlClick(RibbonControl control)
        {
        }
    }
}

File: src/vb/AddIns/AddInDemo/DraftRibbon.vb
Imports SolidEdgeCommunity.AddIn
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace AddInDemo
	Friend Class DraftRibbon
		Inherits SolidEdgeCommunity.AddIn.Ribbon

		Private Const _embeddedResourceName As String = "AddInDemo.DraftRibbon.xml"

		Public Sub New()
			' Get a reference to the current assembly. This is where the ribbon XML is embedded.
			Dim assembly = System.Reflection.Assembly.GetExecutingAssembly()

			' In this example, XML file must have a build action of "Embedded Resource".
			Me.LoadXml(assembly, _embeddedResourceName)
		End Sub

		Public Overrides Sub OnControlClick(ByVal control As RibbonControl)
		End Sub
	End Class
End Namespace
